src/client/game.cpp

```diff
@@ -3641,7 +3661,7 @@ bool Game::nodePlacement(const ItemDefinition &selected_def,
 
 	// formspec in meta
 	if (meta && !meta->getString("formspec").empty() && !input->isRandom()
-			&& !isKeyDown(KeyType::SNEAK)) {
+			&& !(isKeyDown(KeyType::SNEAK) || g_settings->getBool("sneak_lock"))) {
```

Comment:
Can't you use `control.sneak` here?

Reply:
Yes. Idk why I didn't.

Reply:
wait no, looking at that I dont think I can as I would need to make a new PlayerControl object

Reply:
> I would need to make a new PlayerControl object

I'm not suggesting that. Is there no easy way to get the (up-to-date) control object used above? I'm not fond of duplicating control logic here (even though it may be minor).